### PR TITLE
Refactor packaging

### PR DESCRIPTION
**Pull request by ragingpastry**
This is a big one. I am proposing that we change the way we handle packaging and dependency configuration. This PR basically follows how the Ansible Molecule project is doing it.

### CODE REVIEW

**Review by JCantu248 — APPROVED** [review]

**Comment by webknjaz** [comment]
Hey @ragingpastry,

I'm thrilled that you copy my packaging work but FTR since you seem to require Python 3.7 you probably don't need most of the sorcery in `setup.py` since it ships up-to-date enough setuptools capable of reading `setup.cfg`: https://github.com/python/cpython/tree/v3.7.0/Lib/ensurepip/_bundled.

`import setuptools; setuptools.setup()` should do. The only extra thing needed is probably `setuptools-scm` which won't be picked up from `setup.cfg`.

That's why all you need is probably just (more or less):
```python
#! /usr/bin/env python3

__name__ == '__main__' and setuptools.setup(use_scm_version=True, setup_requires=['setuptools-scm'])
```

It'll be easier to maintain since you, unlike Molecule, don't need to support extremely ancient installations.

P.S. If you're brave enough and can rely on the recent pip to bring in the bleeding-edge setuptools you can even skip having `setup.py` at all: https://github.com/sanitizers/octomachinery/commit/4b79eb602d8061f99120f781e1c0649a9e470966

File: setup.cfg [hunk 1]
```diff
@@ -1,19 +1,60 @@
...
-maintainer_email = senior.crepe@gmail.com
-summary = PiCli
+author = Nick Wilburn
+author_email = nwilburn@globalinfotek.com
+url = https://github.com/AFCYBER-DREAM/piedpiper-picli
+maintainer = AFCYBER DREAM Team
+maintainer_email = dream@globalinfotek.com
+summary = A pipeline validation and execution framework
 license = BSD
 description-file = README.md
```
Comment (webknjaz) [inline setup.cfg:10]: this identifier is pbr-specific. it won't work.

File: setup.cfg [hunk 2]
```diff
@@ -1,19 +1,60 @@
...
-home-page = https://github.com/ragingpastry/picli
-requires-python = >=3.6
+home-page = https://github.com/AFCYBER-DREAM/piedpiper-picli
```
Comment (webknjaz) [inline setup.cfg:11]: it's just `url` now.

File: setup.cfg [hunk 3]
```diff
@@ -1,19 +1,60 @@
...
+
+install_requires =
+    anyconfig==0.9.8
+    colorama==0.4.1
+    click==7.0
+    marshmallow>=3.0.0rc4
+    PyYAML==3.13
+    requests==2.21.0
+
+extras =
```
Comment (webknjaz) [inline setup.cfg:37]: this is not valid `setuptools.setup()` arg

**Comment by ragingpastry** [comment]
Thank you very much for your comments!

**Pull request merged**